### PR TITLE
Bookmarking Feature

### PR DESCRIPTION
**Pull request by aryanrsuri**
- [x] sb feat 2.1 is a bookmarking feature
- [x] + some general updates
- [x] added typescript support

### CODE REVIEW

File: public/Aryan Suri/Sidebar/sidebar.js [hunk 1]
```diff
@@ -640,265 +556,228 @@ async function Sidebar() {
...
+    const TITLE = document.getElementById("titleHolder").innerText;
+    const URL = window.location.href;
+    if (URL) {
+        sessionStorage.setItem("Title", TITLE);
+        sessionStorage.setItem("Bookmark", URL);
+        createBookmarks();
+    }
+    else {
+        console.log("Bookmark created already!");
+    }
```
Comment (Miniland1333) [inline public/Aryan Suri/Sidebar/sidebar.js:774]: This check does not appear to be working and will always return true.

File: public/Aryan Suri/Sidebar/sidebar.js [hunk 2]
```diff
@@ -565,14 +489,13 @@ async function Sidebar() {
...
+            "usage": `<div id="sb4"  class="custom_sidebar">
+	<div class="custom_field">
+		<a onclick = "event.preventDefault(); saveBookmark()" href='#' class='mt-icon-quote'>&nbsp;Save Bookmark</a>
+			<div id="bm-list">
 
-    <div class="custom_field">
-         <a onclick = "event.preventDefault(); attribution()" href='#' class='mt-icon-quote'>&nbsp;Get Page Attribution</a>
-    </div>
+			</div>
+	</div>
```
Comment (Miniland1333) [inline public/Aryan Suri/Sidebar/sidebar.js:498]: It looks like get citation and get attribution have been removed. Is this intentional?

**Review by Miniland1333 — CHANGES_REQUESTED** [review]
Here's my comments:

1. The CSS certainly needs improvement for this bookmark feature.
image
2. The check for preventing duplicates is not working.
3. get attribution and citation have been removed, perhaps unintentionally.
4. It is non-obvious that you can only have one bookmark. I do not know if the solution should be to improve the UI and/or allow for multiple bookmarks.

Let me know if you have any questions!
-Henry

File: public/Aryan Suri/Sidebar/sidebar.js (hunk 1 above)
Comment (aryanrsuri) [reply public/Aryan Suri/Sidebar/sidebar.js:774]: Will fix this

File: public/Aryan Suri/Sidebar/sidebar.js (hunk 2 above)
Comment (aryanrsuri) [reply public/Aryan Suri/Sidebar/sidebar.js:498]: Yes

File: public/Aryan Suri/Citation/citation.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
 
 function buildcite() {
-    let citeDiv = document.createElement("div");
-    $(citeDiv).attr("id", "SB-PC-AD");
-    $('body').append(citeDiv);
+    const sidebar = document.getElementById("sidebarDiv");
+    const citeDiv = document.createElement("div");
+
+    sidebar?.style.display === "none";
```
Comment (Miniland1333) [inline public/Aryan Suri/Citation/citation.ts:6]: Nice use of optional chaining!

**Review by Miniland1333 — APPROVED** [review]
Nothing obviously wrong with this. PR approved.
-Henry

**Pull request merged**